From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: asyasocute <asya_bububu@proton.me>
Date: Tue, 28 Jul 2026 16:08:52 +0700
Subject: Hide channel bottom button in channel preview

---
 .../TelegramUI/Sources/ChatInterfaceStateInputPanels.swift     | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/submodules/TelegramUI/Sources/ChatInterfaceStateInputPanels.swift b/submodules/TelegramUI/Sources/ChatInterfaceStateInputPanels.swift
index 0000000..0000000 100644
--- a/submodules/TelegramUI/Sources/ChatInterfaceStateInputPanels.swift
+++ b/submodules/TelegramUI/Sources/ChatInterfaceStateInputPanels.swift
@@ -374,6 +374,9 @@ func inputPanelForChatPresentationIntefaceState(_ chatPresentationInterfaceState
                 if chatPresentationInterfaceState.interfaceState.editMessage != nil, channel.hasPermission(.editAllMessages) {
                     displayInputTextPanel = true
                 } else if !channel.hasPermission(.sendSomething) || !isMember {
+                    if ClearConfig.hideChannelBottomButton {
+                        return (nil, nil)
+                    }
                     if let currentPanel = (currentPanel as? ChatChannelSubscriberInputPanelNode) ?? (currentSecondaryPanel as? ChatChannelSubscriberInputPanelNode) {
                         return (currentPanel, nil)
                     } else {
