From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: asyasocute <asya_bububu@proton.me>
Date: Tue, 28 Jul 2026 01:25:45 +0700
Subject: Add confirm calls toggle

---
 submodules/TelegramUI/Sources/AccountContext.swift | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/submodules/TelegramUI/Sources/AccountContext.swift b/submodules/TelegramUI/Sources/AccountContext.swift
index 0000000..0000000 100644
--- a/submodules/TelegramUI/Sources/AccountContext.swift
+++ b/submodules/TelegramUI/Sources/AccountContext.swift
@@ -820,6 +820,7 @@ public final class AccountContextImpl: AccountContext {
     }
     
     public func requestCall(peerId: PeerId, isVideo: Bool, completion: @escaping () -> Void) {
+        let makeCall = {
         guard let callResult = self.sharedContext.callManager?.requestCall(context: self, peerId: peerId, isVideo: isVideo, endCurrentIfAny: false) else {
             return
         }
@@ -887,6 +888,19 @@ public final class AccountContextImpl: AccountContext {
         } else {
             completion()
         }
+        }
+        
+        if ClearConfig.confirmCalls {
+            let presentationData = self.sharedContext.currentPresentationData.with { $0 }
+            self.sharedContext.mainWindow?.present(textAlertController(context: self, title: nil, text: isVideo ? "Video call?" : "Call?", actions: [TextAlertAction(type: .genericAction, title: presentationData.strings.Common_No, action: {}), TextAlertAction(type: .destructiveAction, title: presentationData.strings.Common_Yes, action: { [weak self] in
+                guard let _ = self else {
+                    return
+                }
+                makeCall()
+            })]), on: .root)
+        } else {
+            makeCall()
+        }
     }
     
     public func getAppConfigValue(_ key: String) -> Any? {
