From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: asyasocute <asya_bububu@proton.me>
Date: Tue, 28 Jul 2026 01:26:03 +0700
Subject: Disable gallery camera

---
 submodules/MediaPickerUI/Sources/MediaPickerScreen.swift | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/submodules/MediaPickerUI/Sources/MediaPickerScreen.swift b/submodules/MediaPickerUI/Sources/MediaPickerScreen.swift
index 0000000..0000000 100644
--- a/submodules/MediaPickerUI/Sources/MediaPickerScreen.swift
+++ b/submodules/MediaPickerUI/Sources/MediaPickerScreen.swift
@@ -711,7 +711,7 @@ public final class MediaPickerScreenImpl: ViewController, MediaPickerScreen, Att
                 }
             }
             
-            if useLegacyCamera {
+            if useLegacyCamera, !ClearConfig.disableGalleryCamera {
                 let enableAnimations = self.controller?.context.sharedContext.energyUsageSettings.fullTranslucency ?? true
   
                 let cameraView = TGAttachmentCameraView(forSelfPortrait: false, videoModeByDefault: controller.bannedSendPhotos != nil && controller.bannedSendVideos == nil)!
@@ -735,7 +735,7 @@ public final class MediaPickerScreenImpl: ViewController, MediaPickerScreen, Att
                 
                 self.gridNode.scrollView.addSubview(cameraView)
                 self.gridNode.addSubnode(self.cameraActivateAreaNode)
-            } else if useModernCamera, !Camera.isIpad {
+            } else if useModernCamera, !Camera.isIpad, !ClearConfig.disableGalleryCamera {
                 #if !targetEnvironment(simulator)
                 var cameraPosition: Camera.Position = .back
                 if case .assets(nil, .createAvatar) = controller.subject {
