From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: asyasocute <asya_bububu@proton.me>
Date: Tue, 28 Jul 2026 01:25:56 +0700
Subject: Disable scroll to next channel

---
 submodules/TelegramUI/Sources/ChatControllerContentData.swift | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/submodules/TelegramUI/Sources/ChatControllerContentData.swift b/submodules/TelegramUI/Sources/ChatControllerContentData.swift
index 0000000..0000000 100644
--- a/submodules/TelegramUI/Sources/ChatControllerContentData.swift
+++ b/submodules/TelegramUI/Sources/ChatControllerContentData.swift
@@ -1103,7 +1103,7 @@ extension ChatControllerImpl {
                         } else {
                             isRegularChat = true
                         }
-                        if strongSelf.nextChannelToReadDisposable == nil, let peerId = chatLocation.peerId, let customChatNavigationStack {
+                        if strongSelf.nextChannelToReadDisposable == nil, let peerId = chatLocation.peerId, let customChatNavigationStack, !ClearConfig.disableScrollToNextChannel {
                             if let index = customChatNavigationStack.firstIndex(of: peerId), index != customChatNavigationStack.count - 1 {
                                 let nextPeerId = customChatNavigationStack[index + 1]
                                 strongSelf.nextChannelToReadDisposable = (combineLatest(queue: .mainQueue(),
@@ -1147,7 +1147,7 @@ extension ChatControllerImpl {
                                     }
                                 })
                             }
-                        } else if isRegularChat, strongSelf.nextChannelToReadDisposable == nil {
+                        } else if isRegularChat, strongSelf.nextChannelToReadDisposable == nil, !ClearConfig.disableScrollToNextChannel {
                             //TODO:loc optimize
                             let accountPeerId = context.account.peerId
                             strongSelf.nextChannelToReadDisposable = (combineLatest(queue: .mainQueue(),
