From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: asyasocute <asya_bububu@proton.me>
Date: Mon, 27 Jul 2026 05:00:51 +0700
Subject: Rename app display name to Cleargram

---
 Telegram/BUILD | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Telegram/BUILD b/Telegram/BUILD
index 0000000..0000000 100644
--- a/Telegram/BUILD
+++ b/Telegram/BUILD
@@ -1568,11 +1568,11 @@ plist_fragment(
     <key>CFBundleDevelopmentRegion</key>
     <string>en</string>
     <key>CFBundleDisplayName</key>
-    <string>Telegram</string>
+    <string>Cleargram</string>
     <key>CFBundleIdentifier</key>
     <string>{telegram_bundle_id}</string>
     <key>CFBundleName</key>
-    <string>Telegram</string>
+    <string>Cleargram</string>
     <key>CFBundlePackageType</key>
     <string>APPL</string>
     <key>CFBundleSignature</key>
